src/cache/cache_operations.rs

```diff
@@ -0,0 +1,131 @@
+use crate::config::{request_cache_duration, request_error_cache_timeout};
+use crate::utils::errors::ApiResult;
+use serde::Serialize;
+
+pub enum Database {
```

Comment:
Why not use an integer?

Reply:
I wanted to put a name with text in it so we remember what the integer stands anywhere we use it. I think we can assign constants to enums in rust like so: 
```rust
pub enum Database {
    Info = 1,
    Default = 0,
}
```
I just need to double check how it behaves

Reply:
Yeah I see the value of that. And I think with the explicit integer definition it would be very nice